esmvalcore/preprocessor/_regrid_iris_esmf_regrid.py

```diff
@@ -0,0 +1,233 @@
+        if all the contributing elements of data are masked. If no value is
+        given, this will default to 1 for conservative regridding and 0
+        otherwise. Only available for methods 'bilinear' and 'conservative'.
+    use_src_mask:
+        If True, derive a mask from the source cube data,
+        which will tell :mod:`esmpy` which points to ignore. If an array is
+        provided, that will be used.
+        If set to :obj:`None`, it will be set to :obj:`True` for methods
+        ``'bilinear'`` and ``'conservative'`` and to :obj:`False` for method
+        ``'nearest'``. This default may be changed to :obj:`True` for all
+        schemes once `SciTools-incubator/iris-esmf-regrid#368
+        <https://github.com/SciTools-incubator/iris-esmf-regrid/issues/368>`_
+        has been resolved.
+    use_tgt_mask:
+        If True, derive a mask from (first time step) of the target cube,
+        which will tell :mod:`esmpy` which points to ignore. If an array is
+        provided, that will be used.
+        If set to :obj:`None`, it will be set to :obj:`True` for methods
+        ``'bilinear'`` and ``'conservative'`` and to :obj:`False` for method
+        ``'nearest'``. This default may be changed to :obj:`True` for all
+        schemes once `SciTools-incubator/iris-esmf-regrid#368`_ has been
+        resolved.
+    collapse_src_mask_along_axes:
+        When deriving the mask from the source cube data, collapse the mask
+        along the dimensions identified by these axes. Only points that are
+        masked at all time (``'T'``), vertical levels (``'Z'``), or both time
+        and vertical levels (``'TZ'``) will be considered masked.
+    collapse_tgt_mask_along_axes:
+        When deriving the mask from the target cube data, collapse the mask
+        along the dimensions identified by these axes. Only points that are
+        masked at all time (``'T'``), vertical levels (``'Z'``), or both time
+        and vertical levels (``'TZ'``) will be considered masked.
+    src_resolution:
+        If present, represents the amount of latitude slices per source cell
+        given to ESMF for calculation. If resolution is set, the source cube
+        must have strictly increasing bounds (bounds may be transposed
+        plus or minus 360 degrees to make the bounds strictly increasing).
+        Only available for method 'conservative'.
+    tgt_resolution:
+        If present, represents the amount of latitude slices per target cell
+        given to ESMF for calculation. If resolution is set, the target cube
+        must have strictly increasing bounds (bounds may be transposed
+        plus or minus 360 degrees to make the bounds strictly increasing).
+        Only available for method 'conservative'.
+    tgt_location:
+        Either "face" or "node". Describes the location for data on the mesh
+        if the target is not a :class:`~iris.cube.Cube`.
```

Comment:
What happens if `tgt_location` is `None` for meshes?

Reply:
It looks like an exception will be raised: https://github.com/SciTools-incubator/iris-esmf-regrid/blob/9e2559af148b0e5ac057fcfeacac3a7ec3e18739/esmf_regrid/schemes.py#L498-L523

Reply:
Ah, I was briefly confused why I didn't get an error in my tests when using the default scheme (which uses `tgt_location=None`), but since I used [unstructured->regular regridding](https://github.com/SciTools-incubator/iris-esmf-regrid/blob/9e2559af148b0e5ac057fcfeacac3a7ec3e18739/esmf_regrid/schemes.py#L1355), this keyword is ignored. I think it would be good to mention that this is required if target is a mesh, but ignored otherwise.

Reply:
Improved in a848468